packages/block-editor/src/components/media-replace-flow/index.js

```diff
@@ -77,8 +77,8 @@ const MediaReplaceFlow = ( {
 	};
 
 	const selectMedia = ( media ) => {
-		onSelect( media );
 		setMediaURLValue( media.url );
+		onSelect( media );
```

Comment:
Could you leave an inline comment? In what case does it unmount the component?

Reply:
I think that depends on the parent component and what it does inside the `onSelect` callback.

The "Post Featured Image" edit component will re-render it when new media data is fetched:
https://github.com/WordPress/gutenberg/blob/7791d58cf790f275a754e21bdeb19109a8129682/packages/block-library/src/post-featured-image/edit.js#L122-L133

Reply:
@ellatrix, the cause of the unmounting can be different in each case, and I don't think the inline comment would be enough to provide the necessary information.

Reply:
Surely there could be a comment explaining why this order is needed? This change is just really easy to revert without getting an error for the automated tests, so it seems like a good place to comment. Also, why are we not getting an error in the automated tests? :)

Reply:
Let's check with #34730.

Reply:
Added inline comment in 55806ce37f8163006756b43a221165f77f2a8943.

Would you please let me know if you think that wording needs changing?

Reply:
Hi, @ellatrix

I just wanted to check if it's okay to merge this fix after adding the inline comment.

Thanks